src/network/connection.js

```diff
@@ -195,8 +195,25 @@ module.exports = class Connection {
 
     this.logDebug('disconnecting...')
     this.connected = false
-    this.socket.end()
     this.socket.unref()
+
+    await new Promise((resolve, reject) => {
+      const timeout = setTimeout(
```

Comment:
`setTimeout` is bad; that's the reason the request timeout is disabled by default. CPU will rise, and the process will be unresponsive depending on the number of close operations.

Reply:
Don't worry, this PR will not be merged. The current solution is not an option. I just opened the PR to start a discussion.